### PR TITLE
features/search_bar_input_54

### PR DESCRIPTION
**Pull request by mballeza**
Commits were squashed into one commit to make future rebasing easier.

Trello description:
"Enhance the search bar by allowing it to take in user inputted data and then pass the data along to the call to the backend REST API endpoint."

### CODE REVIEW

**Comment by iandusenbury** [comment]
Is this still a WIP?

File: client/src/components/app/index.js
Comment (iandusenbury) [inline client/src/components/app/index.js]: What's the thought process on the refresh? We should just be updating the state with the results from the search. This will update the tables automatically.

**Review by halalak — COMMENTED** [review]
Changes look good to me, just need to rebase again before you merge, it seems like the branch still has some conflicts

**Comment by mballeza** [comment]
Note: this is not a WIP. Sorry for any confusion.

**Comment by mballeza** [comment]
I'm not sure what happened when I rebased. View this link to see changes between the current dev commit and my changes: https://github.com/iandusenbury/tableland/pull/43/commits/b55cc6060e58cfb2598d63d1da1a43134a1297f7

**Review by djarredon — APPROVED** [review]
Looks good.

File: client/src/actions/index.js
Comment (iandusenbury) [inline client/src/actions/index.js]: I think this would make more sense to be called `fetchResults`. That's what it's doing right?

File: client/src/containers/navbar/index.js
Comment (iandusenbury) [inline client/src/containers/navbar/index.js]: again, refresh shouldn't be necessary

**Review by iandusenbury — CHANGES_REQUESTED** [review]

File: client/src/actions/index.js
Comment (mballeza) [reply client/src/actions/index.js]: Yeah I can change this.

File: client/src/components/app/index.js
Comment (mballeza) [reply client/src/components/app/index.js]: I will review another option that doesn't require this refresh. The issue I was having was combining the search button press to do two things: navigate to results page and do the api call. I was unclear of how to use the callback functions with the callApi(), but I think now I can achieve it now that we went through an example together working through the roadmap stuff.

File: client/src/components/navbar/navbarForm.js
Comment (halalak) [inline client/src/components/navbar/navbarForm.js]: should this onSubmit prop have the same name as the prop name in the component itself (handleSubmit)?

File: client/src/components/navbar/index.js
Comment (halalak) [inline client/src/components/navbar/index.js]: should this onSubmit prop have the same name as the prop name in the component itself (handleSubmit)?

Comment (mballeza) [reply client/src/components/navbar/index.js]: The handleSubmit prop is a part of the redux form API: https://redux-form.com/6.6.3/docs/api/props.md/

**Comment by mballeza** [comment]
Changes have been made from the comments I received and is ready for another review.

**Review by halalak — APPROVED** [review]

File: client/src/components/navbar/index.js
Comment (iandusenbury) [inline client/src/components/navbar/index.js]: The naming of this component is confusing because in navbarForm.js We're exporting SearchBar.

File: client/src/components/navbar/navbarForm.js
Comment (iandusenbury) [inline client/src/components/navbar/navbarForm.js]: We should create a container for the form instead of passing fetchResults from the navbar

File: client/src/constants/actionTypes.js
Comment (iandusenbury) [inline client/src/constants/actionTypes.js]: CLEAR_MESSAGE and ADD_MESSAGE can be deleted. If you don't you'll have a conflict on this file when merging

**Review by iandusenbury — CHANGES_REQUESTED** [review]

File: client/src/constants/actionTypes.js
Comment (iandusenbury) [reply client/src/constants/actionTypes.js]: Well, maybe I lied. These do need to be deleted though

**Comment by mballeza** [comment]
Changes have been made and is ready for another review.

File: client/src/actions/index.js
Comment (iandusenbury) [inline client/src/actions/index.js]: delete the curly brackets around the dispatch function

`return dispatch =>`

Comment (iandusenbury) [reply client/src/actions/index.js]: If you don't then this won't return a promise

File: client/src/containers/navbar/index.js
Comment (iandusenbury) [inline client/src/containers/navbar/index.js]: Did you do this or was it a result of a rebase?

**Review by iandusenbury — CHANGES_REQUESTED** [review]

File: client/src/containers/navbar/index.js
Comment (mballeza) [reply client/src/containers/navbar/index.js]: Result of a merge.

**Comment by mballeza** [comment]
Removed the curly brackets and switched back the order of actions to resolve the conflict.

**Review by iandusenbury — APPROVED** [review]

**Pull request merged**
